src/services/codefixes/returnValueCorrect.ts

```diff
@@ -0,0 +1,211 @@
+        switch (errorCode) {
+            case Diagnostics.A_function_whose_declared_type_is_neither_void_nor_any_must_return_a_value.code:
+                if (!declaration || !declaration.body || !declaration.type || !rangeContainsRange(declaration.type, node)) return undefined;
+                return getFixInfo(checker, declaration, checker.getTypeFromTypeNode(declaration.type), /* isFunctionType */ false);
+            case Diagnostics.Argument_of_type_0_is_not_assignable_to_parameter_of_type_1.code:
+                if (!declaration || !isCallExpression(declaration.parent) || !declaration.body) return undefined;
+                const pos = declaration.parent.arguments.indexOf(<Expression>declaration);
+                const type = checker.getContextualTypeForArgumentAtIndex(declaration.parent, pos);
+                if (!type) return undefined;
+                return getFixInfo(checker, declaration, type, /* isFunctionType */ true);
+            case Diagnostics.Type_0_is_not_assignable_to_type_1.code:
+                if (!isDeclarationName(node) || !isVariableLike(node.parent) && !isJsxAttribute(node.parent)) return undefined;
+                const initializer = getVariableLikeInitializer(node.parent);
+                if (!initializer || !isFunctionLikeDeclaration(initializer) || !initializer.body) return undefined;
+                return getFixInfo(checker, initializer, checker.getTypeAtLocation(node.parent), /* isFunctionType */ true);
+        }
+        return undefined;
+    }
+
+    function getVariableLikeInitializer(declaration: VariableLikeDeclaration): Expression | undefined {
+        switch (declaration.kind) {
+            case SyntaxKind.VariableDeclaration:
+            case SyntaxKind.Parameter:
+            case SyntaxKind.BindingElement:
+            case SyntaxKind.PropertyDeclaration:
+            case SyntaxKind.PropertyAssignment:
+                return declaration.initializer;
+            case SyntaxKind.JsxAttribute:
+                return declaration.initializer && (isJsxExpression(declaration.initializer) ? declaration.initializer.expression : undefined);
+            case SyntaxKind.ShorthandPropertyAssignment:
+            case SyntaxKind.PropertySignature:
+            case SyntaxKind.EnumMember:
+            case SyntaxKind.JSDocPropertyTag:
+            case SyntaxKind.JSDocParameterTag:
+                return undefined;
+        }
+    }
+
+    function addReturnStatement(changes: textChanges.ChangeTracker, sourceFile: SourceFile, expression: Expression, statement: Statement) {
+        changes.replaceNode(sourceFile, statement, createReturn(expression));
+    }
+
+    function removeBlockBodyBrace(changes: textChanges.ChangeTracker, sourceFile: SourceFile, declaration: ArrowFunction, expression: Expression, withParen: boolean) {
+        changes.replaceNode(sourceFile, declaration.body, (withParen || needsParentheses(expression)) ? createParen(expression) : expression);

```

Comment:
Is this going to delete comments?  I'd be worried about anything between the end of the expression and the closing brace (and maybe things before the opening brace).

Reply:
Looks like it might delete `{/*this comment*/1}`.  Not the end of the world.